ocaml/backends/rtl.ml

```diff
@@ -388,6 +388,14 @@ module Verilog : RTLBackend = struct
     | EUnop (f, e) ->
        (match f with
         | Not _ -> sp "~%a" p e
+        | Rev sz ->
+          let rec loop n = if n = 0
+            then sprintf "%a[%d]" p e n
+            else sprintf "%a[%d], " p e n ^ loop (n - 1) in
```

Comment:
Should we use `sp` here? It looks like it controls operator precedence and adds proper parenthesis.

Reply:
so basically `sp` is defined as:
```
let sp fmt = ksprintf (fun s -> (lvl, s)) fmt in
```

`ksprintf` does the same as sprint except that it passes it's "output" string as input to its first argument

so basically its the same as
```
let sp fmt = (lvl, sprintf fmt) in
``` 
However this version won't work as additional arguments to sp are not passed to sprintf

so `sp` basically generates a tuple, whereas we only want a string to continue the concatenation (which is why I am only using `sprintf`). The tuple is then generated in the line 398.
So basically `sp` is only syntax sugar for the smaller cases

Reply:
Ah! Nice.